shared/extension/NewScript.hpp

```diff
@@ -0,0 +1,285 @@
+#include <nuclear>
+#include <regex>
+#include <yaml-cpp/yaml.h>
+
+#include "extension/Behaviour.hpp"
+
+#include "message/motion/ServoCommand.hpp"
+
+#include "utility/file/fileutil.hpp"
+#include "utility/input/ServoID.hpp"
+#include "utility/support/hostname.hpp"
+
+
+/**
+ * @author Ysobel Sims
+ */
+
+namespace extension::director {
+
+    using message::motion::ServoCommand;
+    using message::motion::ServoState;
+    using utility::input::ServoID;
+
+    // One frame of a script
+    struct Frame {
+        /// @brief Servo targets for each active servo in the frame
+        struct Target {
+            Target() = default;
+            Target(const ServoID& servo, float pos, float gain, float torque)
+                : id(servo), position(pos), gain(gain), torque(torque) {}
+            Target(const Target& other) = default;
+            Target(Target&& other) noexcept
+                : id(other.id), position(other.position), gain(other.gain), torque(other.torque) {}
+            Target& operator=(const Target& other) = default;
+            Target& operator                       =(Target&& other) noexcept {
+                id       = other.id;
+                position = other.position;
+                gain     = other.gain;
+                torque   = other.torque;
+                return *this;
+            }
+
+            /// @brief The servo ID that this target is for; see ServoID.hpp
+            ServoID id;
+            /// @brief Target position for this servo in this frame
+            float position{0.0f};
+            /// @brief Gain for this servo in this frame
+            float gain{0.0f};
+            /// @brief Torque for this servo in this frame
+            float torque{0.0f};
+        };
+
+        Frame() : duration() {}
+        Frame(const NUClear::clock::duration& dur, std::vector<Target> targets)
+            : duration(dur), targets(std::move(targets)) {}
+
+        /// @brief The time it should take for the servos to move to their targets
+        NUClear::clock::duration duration;
```

Comment:
```suggestion
        NUClear::clock::duration duration{};
```
Initialise it, if you delete the constructor.